src/Symfony/Component/Security/Csrf/TokenStorage/AbstractTokenStorageProxy.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Csrf\TokenStorage;
+
+/**
+ * Forwards calls to another TokenStorageInterface.
+ *
+ * @author Oliver Hoff <oliver@hofff.com>
+ */
+abstract class AbstractTokenStorageProxy implements TokenStorageInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getToken($tokenId)
```

Comment:
I think final should only be used in very special cases when you absolutely rely on the implemented behavior and to keep the internal state of the class clean and valid, and this cannot be provided by subclasses (e.g. not being able to access private properties). Both is not the case here. Imagine someone wants to create a decorater that fixes the token ID to a certain value, using this class as a parent would not be possible with final methods.